semana4/lista-de-exercicios/script.js

```diff
@@ -0,0 +1,459 @@
+
+ /**
+  * Exercícios de lógica de programação
+  * 
+  * 1. Para este exercício considere as seguintes variáveis:
+  * const booleano1 = true
+  * const booleano2 = false
+  * const booleano3 = !booleano2
+  * const booleano4 = !booleano3
+  * 
+  * Sem rodar nenhum código, diga quais são os valores das expressões lógicas abaixo:
+  *  
+  * a) `booleano1 && booleano2 && !booleano4`
+  * RESPOSTA: false
+  * 
+  * b) `(booleano1 && booleano2) || !booleano3`
+  * RESPOSTA: true
+  * 
+  * c) `(booleano2 || booleano3) && (booleano4 || booleano1)`
+  * RESPOSTA: true
+  * 
+  * d) `!(booleano2 && booleano3) || !(booleano1 && booleano3)`
+  * RESPOSTA: true
+  * 
+  * e) `!(booleano1) && !(booleano3) || (!booleano4 && booleano3 && booleano3)`
+  * RESPOSTA: true
+  * 
+  * 2. Você tem que escrever um código que, dado um número **N**, ele imprima (no `console`) os **N** primeiros números pares 
+  * (por exemplo, se **N** for 3, você deve imprimir 0, 2 e 4; se **N** for 5, deve imprimir 0, 2, 4, 6 e 8).  Um colega seu disse 
+  * que já começou esta tarefa, mas não conseguiu terminar. Dê uma olhada no código dele:
+  *
+  * const quantidadeDeNumerosPares
+  * let i = 0
+  * while(i <= quantidadeDeNumerosPares) {
+  *     console.log(i*2)
+  * }
+  *
+  *  Este código funciona? Por quê? Caso não funcione, corrija a implementação dele.
+  * RESPOSTA: O código acima não funciona pois não há valor declarado para "quantidadeDeNumerosPares", e quando este número é declarado
+  * é necessário por um contador "i += 1" dentro do "while" para que não fique em looping infnito. E para finalizar no "console.log" onde
+  * é feito o cálculo para compensar a primeira inicialização do "i" e diminuido 2 para que o resultado se inicie em 0 (zero).
+  * 
+  */
+const quantidadeDeNumerosPares = 5;
+let i = 0
+while(i < quantidadeDeNumerosPares) {
+    i += 1;
+    console.log(i*2-2)
+}
+/**
+ *  3. Vocês lembram de trigonometria? ~~(Oh, não, trigonometria)~~. Relaxem. O exercício é simples, mas mexe com isso. 
+ *  Veja bem: quando nos ensinam triângulos (uma figura de três lados), nós aprendemos como classifica-los dependendo do 
+ *  tamanho de seus lados. Se um triângulo possuir os **três lados iguais**, ele é chamado de "Equilátero". Se possuir, **dois 
+ *  (e somente 2) lados iguais**, diz-se que ele é "Isósceles". Se os **três lados tiverem medidas diferentes**, ele é "Escaleno". 
+ *  Faça uma função que receba como parâmetro os tamanhos dos três lados do triângulo: `a, b, c`  e retorne se ele é equilátero, 
+ *  isósceles ou escaleno.
+ */
+function tipoTriangulo(ladoA, ladoB, ladoC) {
+    if(ladoA === ladoB && ladoA === ladoC) {
+        console.log("Equilátero")
+    } else if(ladoA === ladoB && ladoA !== ladoC || 
+              ladoA === ladoC && ladoA !== ladoB ||
+              ladoB === ladoC && ladoB !== ladoA) {
+        console.log("Isósceles")
+    } else if(ladoA !== ladoB && ladoA !== ladoC && ladoB !== ladoC) {
+        console.log("Escaleno")
```

Comment:
Como vc já tinha feito a verificação do triangulo equilátero na linha 97 e do triangulo isósceles na linha 99..., não precisava colocar a verificação do triangulo escaleno na linha 103. Bastava colocar um else no final. Mas tbm não está errado do jeito que vc fez.